library/spdm_responder_lib/libspdm_rsp_set_certificate.c

```diff
@@ -39,11 +40,22 @@ static bool libspdm_set_cert_verify_certchain(const uint8_t *cert_chain, size_t
         return false;
     }
 
-    /*verify leaf cert*/
-    if (!libspdm_x509_set_cert_certificate_check(leaf_cert_buffer, leaf_cert_buffer_size,
-                                                 base_asym_algo, base_hash_algo,
-                                                 false, is_device_cert_model)) {
-        return false;
+    if (spdm_version == SPDM_MESSAGE_VERSION_12) {
+        const bool is_device_cert_model =
+            (cert_model == SPDM_CERTIFICATE_INFO_CERT_MODEL_DEVICE_CERT);
+
+        /*verify leaf cert*/
+        if (!libspdm_x509_set_cert_certificate_check(leaf_cert_buffer, leaf_cert_buffer_size,
+                                                     base_asym_algo, base_hash_algo,
+                                                     false, is_device_cert_model)) {
+            return false;
+        }
+    } else {
+        if (!libspdm_x509_set_cert_certificate_check_ex(leaf_cert_buffer, leaf_cert_buffer_size,
+                                                        base_asym_algo, base_hash_algo,
+                                                        false, cert_model)) {
```

Comment:
It looks weird to only support 1.3 in ex version and only support 1.2 in non-ex version.
I feel it will be easily mis-used, if it is designed in such way.

But I don't have good idea now. Need to think through.